File: colcon_ws/src/pkg_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

using std::placeholders::_1;

class SimpleSub: public rclcpp::Node
{
public:
    SimpleSub() : Node("simple_subscriber")
    {
        sub_ = this->create_subscription<std_msgs::msg::String>("chatter", 10, std::bind(&SimpleSub::msgCallback, this, _1));
    }

    void msgCallback(const std_msgs::msg::String &msg) const
    {
        RCLCPP_INFO_STREAM(get_logger(), "I heard: " <<  msg.data.c_str());
    }

    
private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;
};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleSub>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}

File: colcon_ws/src/pkg_python_examples/pkg_python_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimpleSub(Node):
    """
    This class implements the example to show how the subscriber
    works in ROS2
    """
    def __init__(self):
        # this is class constuctor
        super().__init__(node_name="simple_subscriber")
        self.sub_ = self.create_subscription(String, "chatter", self.sub_callback, 10)
    
    def sub_callback(self, msg):
        self.get_logger().info(f"I heard: {msg.data}")

def main():
    rclpy.init()
    simple_sub = SimpleSub()
    rclpy.spin(simple_sub)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
